extensions/wikia/Wall/WallMessage.class.php

```diff
@@ -39,24 +42,41 @@ function __construct( Title $title, $articleComment = null ) {
 		wfProfileOut( __METHOD__ );
 	}
 
-	static public function newFromId( $id, $master = false ) {
-		wfProfileIn( __METHOD__ );
+	/**
+	 * @param int $id
+	 * @param bool $master
+	 * @return null|WallMessage
+	 */
+	static public function newFromId( int $id, $master = false ) {
 
-		if ( $master == true ) {
-			$title = Title::newFromId( $id, Title::GAID_FOR_UPDATE );
-		} else {
-			$title = Title::newFromId( $id );
+		// WallMessage::getTopParentObj may call this method with $id set to zero
+		if ( $id === 0 ) {
+			return null;
+		}
+
+		// use in-memory cache
+		if ( !empty( self::$wallMessageCache[$id] ) ) {
+			return self::$wallMessageCache[$id];
 		}
 
+		wfProfileIn( __METHOD__ );
+
+		$title = Title::newFromID( $id, $master == true ? Title::GAID_FOR_UPDATE : 0 );
+
 		if ( $title instanceof Title && $title->exists() ) {
 			wfProfileOut( __METHOD__ );
-			return WallMessage::newFromTitle( $title );
+			return self::$wallMessageCache[$id] = WallMessage::newFromTitle( $title );
 		}
 
 		if ( $master == false ) {
+			// TODO: instead of relying on fallback to master let's implement a proper wfWaitForSlaves() use
```

Comment:
Let's wait for the logs to went live, take a look at them and decide then. Maybe nothing will be logged (fingers crossed ;)).